core/src/main/java/org/apache/carbondata/core/datastore/filesystem/LocalCarbonFile.java

```diff
@@ -162,6 +163,20 @@ public boolean delete() {
     return carbonFiles;
   }
 
+  @Override
+  public CarbonFile[] listFiles(boolean recursive, int maxCount)
+      throws IOException {
+    List<CarbonFile> carbonFiles = new ArrayList<>();
+    int counter = 0;
+    Iterator it = FileUtils.iterateFiles(file, null, recursive);
+    while (it.hasNext() && counter < maxCount) {
+      CarbonFile carbonFile = new LocalCarbonFile((File) it.next());
+      carbonFiles.add(carbonFile);
+      counter++;
+    }
+    return carbonFiles.toArray(new CarbonFile[0]);
```

Comment:
Use a constant to replace new CarbonFile[0]

Reply:
From java 8 on, use `CarbonFile[0]` is better, please check please check https://stackoverflow.com/questions/174093/toarraynew-myclass0-or-toarraynew-myclassmylist-size

Reply:
Use CarbonFile[0] is better than use CarbonFile[carbonFiles.size()].
Is need not new a empty array every time here.